CarpetIOHDF5/src/OutputSlice.cc

```diff
@@ -699,7 +715,19 @@ void IOHDF5<outdim>::OutputDirection(const cGH *const cctkGH, const int vindex,
       c_base += exts.setsize();
     } // for c
 
-    error_count += CloseFile(cctkGH, file, index_file);
+    if (CCTK_EQUALS(flush_to_disk, "immediate"))
+      error_count += CloseFile(cctkGH, file);
+
+    if (nioprocs > 1) {
```

Comment:
@eschnett says

This would do the wrong thing if there are `n` processes, but only `1` I/O process. `MPI_Allreduce` for `dist::comm()` is superfluous only if `nprocs==1`.

Reply:
Not quite sure I follow. If (I quote) “is superfluous only `if nprocs==1`“ and the check is `if (nioprocs > 1)` is the condition used (and nioprocs >= 1) isn’t `if(nioprocs > 1)` the same as `if(nioprocs == 1)`.
I suspect that for this IO thorn there is either `nioprocs == 1` or `nioprocs == nprocs`. In the first case then the ioproc already has “good” values in `io_files` at least that seems to be the assumption in the place where I copied this code from, namely:  
I would think that `AllReduce` can be safe if `io_files` and `io_bytes` are initialized to 0 on al ranks initially. Should have tests.

Reply:
This code looks correct to me, or at least as correct as the it was when it was still part of the `CloseFile` function (which is where it was moved from).